lib/puppet/transaction/report.rb

```diff
@@ -176,7 +178,7 @@ def add_resource_status(status)
 
   # @api private
   def compute_status(resource_metrics, change_metric)
-    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0
+    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0 || @logs.any? { |lg| lg.level == :err }
```

Comment:
This is a side-effect of the report being a log destination. So after we finialize the report, any log output gets written to the report. Also leads to issues where the last saved report on disk may not be the same as what we sent.